File: composeApp/src/androidMain/kotlin/com/prekogdevs/fintrack/feature/dashboard/components/AddTransactionDialog.kt
package com.prekogdevs.fintrack.feature.dashboard.components

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ExposedDropdownMenuBox
import androidx.compose.material3.ExposedDropdownMenuDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog
import com.prekogdevs.fintrack.core.theme.AppColors


@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AddTransactionDialog(onDismiss: () -> Unit) {
    var selectedType by remember { mutableStateOf("Expense") }
    var description by remember { mutableStateOf("") }
    var amount by remember { mutableStateOf("") }
    var selectedCategory by remember { mutableStateOf("Food & Dining") }
    var expanded by remember { mutableStateOf(false) }

    val categories = listOf(
        "Food & Dining", "Transport", "Shopping", "Coffee", "Health", "Income", "Other"
    )

    Dialog(onDismissRequest = onDismiss) {
        Card(
            shape = RoundedCornerShape(24.dp),
            colors = CardDefaults.cardColors(containerColor = AppColors.CardWhite),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.padding(24.dp)) {
                // Header
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        "Add Transaction",
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold,
                        color = AppColors.TextPrimary
                    )
                    IconButton(
                        onClick = onDismiss,
                        modifier = Modifier
                            .size(36.dp)
                            .clip(CircleShape)
                            .background(AppColors.BgGray)
                    ) {
                        Icon(Icons.Default.Close, contentDescription = "Close", tint = AppColors.TextPrimary)
                    }
                }

                Spacer(Modifier.height(20.dp))

                // Type toggle
                Text("Type", fontSize = 13.sp, color = AppColors.TextSecondary, fontWeight = FontWeight.Medium)
                Spacer(Modifier.height(8.dp))
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(10.dp)
                ) {
                    listOf("Expense", "Income").forEach { type ->
                        val isSelected = selectedType == type
                        val bgColor by animateColorAsState(
                            targetValue = when {
                                isSelected && type == "Expense" -> AppColors.RedExpense
                                isSelected && type == "Income"  -> AppColors.GreenIncome
                                else -> AppColors.BgGray
                            },
                            animationSpec = tween(200), label = "typeColor"
                        )
                        Box(
                            modifier = Modifier
                                .weight(1f)
                                .clip(RoundedCornerShape(12.dp))
                                .background(bgColor)
                                .clickable { selectedType = type }
                                .padding(vertical = 14.dp),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(
                                text = type,
                                color = if (isSelected) Color.White else AppColors.TextSecondary,
                                fontWeight = FontWeight.SemiBold,
                                fontSize = 15.sp
                            )
                        }
                    }
                }

                Spacer(Modifier.height(20.dp))

                // Category dropdown
                Text("Category", fontSize = 13.sp, color = AppColors.TextSecondary, fontWeight = FontWeight.Medium)
                Spacer(Modifier.height(8.dp))
                ExposedDropdownMenuBox(
                    expanded = expanded,
                    onExpandedChange = { expanded = !expanded }
                ) {
                    OutlinedTextField(
                        value = selectedCategory,
                        onValueChange = {},
                        readOnly = true,
                        trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded) },
                        modifier = Modifier
                            .fillMaxWidth()
                            .menuAnchor(),
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            unfocusedBorderColor = Color.Transparent,
                            focusedBorderColor = AppColors.BluePrimary,
                            unfocusedContainerColor = AppColors.BgGray,
                            focusedContainerColor = AppColors.BgGray
                        )
                    )
                    ExposedDropdownMenu(
                        expanded = expanded,
                        onDismissRequest = { expanded = false }
                    ) {
                        categories.forEach { cat ->
                            DropdownMenuItem(
                                text = { Text(cat) },
                                onClick = {
                                    selectedCategory = cat
                                    expanded = false
                                }
                            )
                        }
                    }
                }

                Spacer(Modifier.height(16.dp))

                // Description
                Text("Description", fontSize = 13.sp, color = AppColors.TextSecondary, fontWeight = FontWeight.Medium)
                Spacer(Modifier.height(8.dp))
                OutlinedTextField(
                    value = description,
                    onValueChange = { description = it },
                    placeholder = { Text("Enter description", color = AppColors.TextSecondary) },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(12.dp),
                    colors = OutlinedTextFieldDefaults.colors(
                        unfocusedBorderColor = Color.Transparent,
                        focusedBorderColor = AppColors.BluePrimary,
                        unfocusedContainerColor = AppColors.BgGray,
                        focusedContainerColor = AppColors.BgGray
                    )
                )

                Spacer(Modifier.height(16.dp))

                // Amount
                Text("Amount", fontSize = 13.sp, color = AppColors.TextSecondary, fontWeight = FontWeight.Medium)
                Spacer(Modifier.height(8.dp))
                OutlinedTextField(
                    value = amount,
                    onValueChange = { amount = it },
                    placeholder = { Text("0.00", color = AppColors.TextSecondary) },
                    prefix = { Text("$ ", color = AppColors.TextSecondary) },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(12.dp),
                    colors = OutlinedTextFieldDefaults.colors(
                        unfocusedBorderColor = Color.Transparent,
                        focusedBorderColor = AppColors.BluePrimary,
                        unfocusedContainerColor = AppColors.BgGray,
                        focusedContainerColor = AppColors.BgGray
                    )
                )

                Spacer(Modifier.height(24.dp))

                // Submit button
                Button(
                    onClick = onDismiss,
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(52.dp),
                    shape = RoundedCornerShape(14.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = AppColors.BluePrimary)
                ) {
                    Text(
                        "Add Transaction",
                        fontSize = 16.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color.White
                    )
                }
            }
        }
    }
}

File: iosApp/iosApp/Feature/Dashboard/Components/AddTransactionDialog.swift
import SwiftUI
import Shared

struct AddTransactionDialog: View {
    let categories: [Shared.Category]
    let onDismiss: () -> Void
    let onAddTransaction: (Shared.Transaction) -> Void

    @State private var selectedType = "Expense"
    @State private var description = ""
    @State private var amount = ""
    @State private var selectedCategoryName: String

    private static let transactionTypes = ["Expense", "Income"]

    init(
        categories: [Shared.Category],
        onDismiss: @escaping () -> Void,
        onAddTransaction: @escaping (Shared.Transaction) -> Void
    ) {
        self.categories = categories
        self.onDismiss = onDismiss
        self.onAddTransaction = onAddTransaction
        _selectedCategoryName = State(initialValue: categories.first?.name ?? "")
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 20) {
            // Header
            HStack {
                Text("Add Transaction")
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(AppColors.textPrimary)
                Spacer()
                Button(action: onDismiss) {
                    Image(systemName: "xmark")
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(AppColors.textPrimary)
                        .frame(width: 36, height: 36)
                        .background(AppColors.bgGray)
                        .clipShape(Circle())
                }
            }

            // Type toggle
            VStack(alignment: .leading, spacing: 8) {
                Text("Type")
                    .font(.system(size: 13, weight: .medium))
                    .foregroundColor(AppColors.textSecondary)

                HStack(spacing: 10) {
                    ForEach(Self.transactionTypes, id: \.self) { type in
                        let isSelected = selectedType == type
                        Button(action: { withAnimation(.easeInOut(duration: 0.2)) { selectedType = type } }) {
                            Text(type)
                                .font(.system(size: 15, weight: .semibold))
                                .foregroundColor(isSelected ? .white : AppColors.textSecondary)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 14)
                                .background(typeBackground(for: type, isSelected: isSelected))
                                .cornerRadius(12)
                        }
                    }
                }
            }

            // Category picker
            VStack(alignment: .leading, spacing: 8) {
                Text("Category")
                    .font(.system(size: 13, weight: .medium))
                    .foregroundColor(AppColors.textSecondary)

                Picker("Category", selection: $selectedCategoryName) {
                    ForEach(categories) { cat in
                        Text(cat.name).tag(cat.name)
                    }
                }
                .pickerStyle(.menu)
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.horizontal, 12)
                .frame(height: 52)
                .background(AppColors.bgGray)
                .cornerRadius(12)
            }

            // Description
            VStack(alignment: .leading, spacing: 8) {
                Text("Description")
                    .font(.system(size: 13, weight: .medium))
                    .foregroundColor(AppColors.textSecondary)

                TextField("Enter description", text: $description)
                    .padding(.horizontal, 12)
                    .frame(height: 52)
                    .background(AppColors.bgGray)
                    .cornerRadius(12)
            }

            // Amount
            VStack(alignment: .leading, spacing: 8) {
                Text("Amount")
                    .font(.system(size: 13, weight: .medium))
                    .foregroundColor(AppColors.textSecondary)

                HStack {
                    Text("$")
                        .foregroundColor(AppColors.textSecondary)
                    TextField("0.00", text: $amount)
                        .keyboardType(.decimalPad)
                }
                .padding(.horizontal, 12)
                .frame(height: 52)
                .background(AppColors.bgGray)
                .cornerRadius(12)
            }

            // Submit button
            Button(action: {
                if let parsedAmount = Double(amount), !description.isEmpty, parsedAmount > 0 {
                    let type: TransactionType = selectedType == "Expense" ? .expense : .income
                    onAddTransaction(Shared.Transaction(
                        id: "",
                        amount: parsedAmount,
                        description: description,
                        category: selectedCategoryName,
                        type: type,
                        date: "",
                        currency: "USD"
                    ))
                }
            }) {
                Text("Add Transaction")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 52)
                    .background(AppColors.bluePrimary)
                    .cornerRadius(14)
            }
        }
        .padding(24)
        .background(AppColors.cardWhite)
    }

    private func typeBackground(for type: String, isSelected: Bool) -> Color {
        guard isSelected else { return AppColors.bgGray }
        return type == "Expense" ? AppColors.redExpense : AppColors.greenIncome
    }
}
